src/threading/index.mjs

```diff
@@ -56,7 +56,11 @@ export default class WorkerPool {
           this.changeActiveThreadCount(-1);
           this.processQueue(threads);
 
-          (result?.error ? reject : resolve)(result);
+          if (result?.error) {
```

Comment:
Why change this? :eyes:

Reply:
If `result.error` exists, we should call `reject(result.error)`. Otherwise, we call `resolve(result)`. The previous implementation used `reject(result)`, which would wrap the actual error inside an object like `{ error: [Actual Error] }` instead of passing the error directly.